### PR TITLE
Add missing opt-in annotations

### PR DESCRIPTION
**Pull request by 3flex**

### CODE REVIEW

File: poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/equalsGeneration.kt [hunk 1]
```diff
@@ -214,6 +215,7 @@ private fun IrBuilderWithScope.irCallContentDeepEquals(
  * function if it represents a primitive array.
  */
 context(IrPluginContext)
+@OptIn(UnsafeDuringIrConstructionAPI::class)
```
Comment (JakeWharton) [inline poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/equalsGeneration.kt:218]:
```suggestion
@OptIn(UnsafeDuringIrConstructionAPI::class)
context(IrPluginContext)
```
In my brain the context is part of the signature

File: poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/hashCodeGeneration.kt [hunk 2]
```diff
@@ -140,6 +141,7 @@ private fun IrBlockBodyBuilder.getHashCodeOfProperty(
  * [org.jetbrains.kotlin.fir.backend.generators.DataClassMembersGenerator].
  */
 context(IrPluginContext)
+@OptIn(UnsafeDuringIrConstructionAPI::class)
```
Comment (JakeWharton) [inline poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/hashCodeGeneration.kt:144]:
```suggestion
@OptIn(UnsafeDuringIrConstructionAPI::class)
context(IrPluginContext)
```

File: poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/hashCodeGeneration.kt [hunk 3]
```diff
@@ -264,6 +266,7 @@ private fun IrBlockBodyBuilder.maybeFindArrayContentHashCodeFunction(
 }
 
 context(IrPluginContext)
+@OptIn(UnsafeDuringIrConstructionAPI::class)
```
Comment (JakeWharton) [inline poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/hashCodeGeneration.kt:269]:
```suggestion
@OptIn(UnsafeDuringIrConstructionAPI::class)
context(IrPluginContext)
```

File: poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/toStringGeneration.kt [hunk 4]
```diff
@@ -186,6 +187,7 @@ private fun IrBlockBodyBuilder.irArrayTypeCheckAndContentDeepToStringBranch(
  * `contentToString` function if it is a primitive array.
  */
 context(IrPluginContext)
+@OptIn(UnsafeDuringIrConstructionAPI::class)
```
Comment (JakeWharton) [inline poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/toStringGeneration.kt:190]:
```suggestion
@OptIn(UnsafeDuringIrConstructionAPI::class)
context(IrPluginContext)
```

**Review by JakeWharton — APPROVED** [review]

File: poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/equalsGeneration.kt (hunk 1 above)
Comment (3flex) [reply poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/equalsGeneration.kt:218]: Makes sense - I just tried though and there's a compiler error:

image

Comment (drewhamilton) [reply poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/equalsGeneration.kt:218]: Let this awkward syntax be a reminder to remove context receivers from this project 😬

**Pull request merged**